File: db/event_related.php
<?php

require_once("entity.php");

/**
 * Class for accessing, manipulating, creating and deleting records in event_related.
*/

class event_related extends Entity
{

  public function __construct($select = array())
  {
    $this->table = "event_related";
    $this->order = "";
    $this->domain = "event";

    $this->field['event_id1']['type'] = 'INTEGER';
    $this->field['event_id1']['not_null'] = true;
    $this->field['event_id2']['type'] = 'INTEGER';
    $this->field['event_id2']['not_null'] = true;
    $this->field['event_id1']['primary_key'] = true;
    $this->field['event_id2']['primary_key'] = true;
    parent::__construct($select);
  }

}

?>

File: pentabarf/lib/momomoto/tables/event_related.rb
module Momomoto
  class Event_related < Base
    def initialize
      super
      @domain = 'event'
      @fields = {
        :event_id1 => Datatype::Integer.new( {:not_null=>true, :primary_key=>true} ),
        :event_id2 => Datatype::Integer.new( {:not_null=>true, :primary_key=>true} )
      }
    end
  end
end
